src/components/PeoplePage.tsx

```diff
@@ -1,31 +1,138 @@
+  const [error, setError] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+
+  // const [people, setPeople] = useState(peopleFromServer);
+  const [searchParams, setSearchParams] = useSearchParams();
+  const sort = searchParams.get('sort');
+  const order = searchParams.get('order');
+  const sex = searchParams.get('sex');
+  const query = searchParams.get('query');
+  const centuries = searchParams.getAll('centuries');
+
+  // #region initialLoad
+  const getPeopleFromServer = async () => {
+    try {
+      setIsLoading(true);
+      const response = await getPeople();
+
+      setPeopleFromServer(response);
+    } catch {
+      setError(true);
+      setTimeout(() => setError(false), 1000);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    getPeopleFromServer();
+  }, []);
+  // #endregion
+
+  // #region getVisiblePeople
+
+  let visiblePeople = [...peopleFromServer];
+
+  // #region filter
+  if (sex) {
+    visiblePeople = visiblePeople.filter(person => person.sex === sex);
+  }
+
+  if (query) {
+    const ifIncludes = (personName: string | null) => {
+      return personName && personName.toLowerCase().includes(query);
+    };
+
+    visiblePeople = visiblePeople.filter(
+      ({ name, motherName, fatherName }) => ifIncludes(name)
+        || ifIncludes(motherName)
+        || ifIncludes(fatherName),
+    );
+  }
+
+  if (centuries.length) {
+    visiblePeople = visiblePeople.filter(
+      person => centuries.includes(Math.ceil(person.born / 100).toString()),
+    );
+  }
+  // #endregion
+
+  // #region sort
+  if (sort) {
+    visiblePeople.sort((a, b) => {
+      switch (sort) {
+        case 'name':
+        case 'sex':
+          return a[sort].localeCompare(b[sort]);
+        case 'born':
+        case 'died':
+          return (a[sort] - b[sort]);
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (order) {
+    visiblePeople = visiblePeople.reverse();
+  }
+  // #endregion
+
+  // #endregion
+
   return (
     <>
       <h1 className="title">People Page</h1>
 
       <div className="block">
         <div className="columns is-desktop is-flex-direction-row-reverse">
           <div className="column is-7-tablet is-narrow-desktop">
-            <PeopleFilters />
+            {!isLoading && (
+              <PeopleFilters
+                searchParams={searchParams}
+                setSearchParams={setSearchParams}
+              />
+            )}
           </div>
 
           <div className="column">
             <div className="box table-container">
-              <Loader />
-
-              <p data-cy="peopleLoadingError">Something went wrong</p>
-
-              <p data-cy="noPeopleMessage">
-                There are no people on the server
-              </p>
-
-              <p>There are no people matching the current search criteria</p>
-
-              <PeopleTable />
+              {(() => {
```

Comment:
Immediately invoked function with switch case using directly in JSX decrease readability. It would be better to use conditional rendering here. It is much more simplify and readable